pkg/serviceimport/weights/controller_test.go

```diff
@@ -0,0 +1,167 @@
+)
+
+type weightMapTestDriver struct {
+	controller *weights.Controller
+	kubeClient lighthouseClientset.Interface
+}
+
+func createDefaultWeightMap(name string) *lighthousev1a1.ServiceImportWeightMap {
+	return &lighthousev1a1.ServiceImportWeightMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: lighthousev1a1.ServiceImportWeightMapSpec{
+
+			SourceClusterWeightMap: map[string]*lighthousev1a1.ClusterWeightMap{
+				cluster1: {
+					NamespaceWeightMap: map[string]*lighthousev1a1.NamespaceWeightMap{
+						ns1: {
+							ServiceWeightMap: map[string]*lighthousev1a1.ServiceWeightMap{
+								scv1: {
+									TargetClusterWeightMap: map[string]int64{
+										cluster2: 8,
+										cluster3: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (n *weightMapTestDriver) updateWeightsMap(name, service, namesapce, inCluster string, weight int64) {
+	weightMapObj := createDefaultWeightMap(name)
+	weightMapObj.
+		Spec.
+		SourceClusterWeightMap[cluster1].
+		NamespaceWeightMap[namesapce].
+		ServiceWeightMap[service].
+		TargetClusterWeightMap[inCluster] = weight
+
+	_, err := n.kubeClient.LighthouseV1alpha1().
+		ServiceImportWeightMaps(metav1.NamespaceAll).
+		Update(context.TODO(), weightMapObj, metav1.UpdateOptions{})
+	Expect(err).To(Succeed())
+}
+
+func (n *weightMapTestDriver) createWeightsMap(name string) {
+	weightMapObj := createDefaultWeightMap(name)
+
+	_, err := n.kubeClient.LighthouseV1alpha1().
+		ServiceImportWeightMaps(metav1.NamespaceAll).
+		Create(context.TODO(), weightMapObj, metav1.CreateOptions{})
+	Expect(err).To(Succeed())
+}
+
+func (n *weightMapTestDriver) deleteWeightsMap(name string) {
+	err := n.kubeClient.LighthouseV1alpha1().ServiceImportWeightMaps(metav1.NamespaceAll).Delete(context.TODO(), name, metav1.DeleteOptions{})
+	Expect(err).To(Succeed())
+}
+
+func newWeightMapTestDiver() *weightMapTestDriver {
+	t := &weightMapTestDriver{}
+
+	BeforeEach(func() {
+		t.kubeClient = fakeClientSet.NewSimpleClientset()
+		t.controller = weights.NewController(cluster1)
+		t.controller.NewClientset = func(c *rest.Config) (lighthouseClientset.Interface, error) {
+			return t.kubeClient, nil
+		}
+
+		Expect(t.controller.Start(&rest.Config{})).To(Succeed())
+		Expect(v1.AddToScheme(scheme.Scheme)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		t.controller.Stop()
+	})
+
+	return t
+}
+
+var _ = Describe("ServiceImportWeights controller", func() {
+	t := newWeightMapTestDiver()
+
+	validateWeightAvailable := func(service, namesapce, inCluster string, weight int64) {
+		Eventually(func() int64 {
+			return t.controller.GetWeightFor(service, namesapce, inCluster)
+		}, 5).Should(Equal(weight))
+	}
+
+	validateWeightNotAvailable := func(service, namesapce, inCluster string) {
+		Eventually(func() int64 {
+			return t.controller.GetWeightFor(service, namesapce, inCluster)
+		}, 5).Should(Equal(int64(1)))
+	}
+
+	When("weigth does not exists for query", func() {
```

Comment:
_weigth_ (sic ) - same in several places below as well.

```suggestion
	When("no weight is defined", func() {
```